splunk/src/main/java/com/splunk/HttpService.java

```diff
@@ -36,8 +36,17 @@
 public class HttpService {
     // For debugging purposes
     private static final boolean VERBOSE_REQUESTS = false;
-    public static boolean useTLS=false;
+    public static boolean useTLS = false;
     protected static SSLSecurityProtocol sslSecurityProtocol = null;
+
+    /**
+     * Boolean flag for validating certificates at either of the sides (client/server).
+     * If true, then it will check and validate relevant certificates otherwise, in case of false, it will accept all certificates.
+     * For PROD environment, TRUE is strongly recommended, whereas working in localhost OR development environment, FALSE is used.
+     * Default Value: TRUE
+     */
+    public static boolean validateCertificates = true;

```

Comment:
Yes. This was implemented earlier in createSSLFactory method based on the validateCertificate flag. 
I think this would be more appropriate for both scenarios. (w/ certificate & w/o certificate).